airflow-core/src/airflow/ui/src/pages/XCom/XComFilters.tsx

```diff
@@ -0,0 +1,180 @@
+
+const FILTERS = [
+  {
+    hotkeyDisabled: false,
+    key: SearchParamsKeys.KEY_PATTERN,
+    translationKey: "keyPlaceholder",
+    type: "search",
+  },
+  {
+    hotkeyDisabled: true,
+    key: SearchParamsKeys.DAG_DISPLAY_NAME_PATTERN,
+    translationKey: "dagDisplayNamePlaceholder",
+    type: "search",
+  },
+  {
+    hotkeyDisabled: true,
+    key: SearchParamsKeys.RUN_ID_PATTERN,
+    translationKey: "runIdPlaceholder",
+    type: "search",
+  },
+  {
+    hotkeyDisabled: true,
+    key: SearchParamsKeys.TASK_ID_PATTERN,
+    translationKey: "taskIdPlaceholder",
+    type: "search",
+  },
+  {
+    hotkeyDisabled: true,
+    key: SearchParamsKeys.MAP_INDEX,
+    translationKey: "mapIndexPlaceholder",
+    type: "search",
+  },
+  {
+    key: SearchParamsKeys.LOGICAL_DATE_GTE,
+    translationKey: "logicalDateFromPlaceholder",
+    type: "datetime",
+  },
+  {
+    key: SearchParamsKeys.LOGICAL_DATE_LTE,
+    translationKey: "logicalDateToPlaceholder",
+    type: "datetime",
+  },
+  {
+    key: SearchParamsKeys.RUN_AFTER_GTE,
+    translationKey: "runAfterFromPlaceholder",
+    type: "datetime",
+  },
+  {
+    key: SearchParamsKeys.RUN_AFTER_LTE,
+    translationKey: "runAfterToPlaceholder",
+    type: "datetime",
+  },
+] as const;
+
+export const XComFilters = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const { setTableURLState, tableURLState } = useTableURLState();
+  const { pagination, sorting } = tableURLState;
+  const { t: translate } = useTranslation(["browse", "common"]);
+  const [resetKey, setResetKey] = useState(0);
+
+  const handleFilterChange = useCallback(
+    (paramKey: string) => (value: string) => {
+      if (value === "") {
+        searchParams.delete(paramKey);
+      } else {
+        searchParams.set(paramKey, value);
+      }
+      setTableURLState({
+        pagination: { ...pagination, pageIndex: 0 },
+        sorting,
+      });
+      setSearchParams(searchParams);
+    },
+    [pagination, searchParams, setSearchParams, setTableURLState, sorting],
+  );
+
+  const filterCount = useMemo(
+    () =>
+      FILTERS.filter((filter) => {
+        const value = searchParams.get(filter.key);
+
+        return value !== null && value !== "";
+      }).length,
+    [searchParams],
+  );
+
+  const handleResetFilters = useCallback(() => {
+    FILTERS.forEach((filter) => {
+      searchParams.delete(filter.key);
+    });
+    setTableURLState({
+      pagination: { ...pagination, pageIndex: 0 },
+      sorting,
+    });
+    setSearchParams(searchParams);
+    setResetKey((prev) => prev + 1);
+  }, [pagination, searchParams, setSearchParams, setTableURLState, sorting]);
+
+  const renderFilterInput = (filter: (typeof FILTERS)[number]) => {
+    const { key, translationKey, type } = filter;
+
+    return (
+      <Box key={key} w="200px">
+        <Text fontSize="xs" marginBottom={1}>
+          {type === "search" ? "\u00A0" : translate(`common:filters.${translationKey}`)}
```

Comment:
```suggestion
          {type === "search" ? "\u00A0" : translate(`common:filters.${translationKey}`)}
```

Can you explain `\u00A0`?

Reply:
Using `\u00A0` ensures the search filter container has the same height as those time-related filters (e.g., "Logical Date from"). Without it, the search box would be higher than the rest. Like this:

<img width="2134" height="368" alt="filter" src="https://github.com/user-attachments/assets/fdb85622-e920-4e6b-8104-d743522c8e98" />

Reply:
This can be achieved via CSS and we should prefer this because it's easier to understand.

Reply:
Sure, I'll try.